pkg/api/volumes.go

```diff
@@ -61,11 +61,20 @@ func UpdateVolumeObject(target *common.VolumeObject, source *client.VolumesResou
 }
 
 // CreateVolume : creates a volume with the given name, capacity in the given pool
-func (client *Client) CreateVolume(name, size, pool, poolType string) (*common.VolumeObject, *common.ResponseStatus, error) {
+func (myclient *Client) CreateVolume(name, size, pool, poolType string) (*common.VolumeObject, *common.ResponseStatus, error) {
 
-	logger := klog.FromContext(client.Ctx)
-	response, httpRes, err := client.apiClient.DefaultApi.CreateVolumePoolSizeTierAffinityNameGet(client.Ctx, pool, size, ApiTierAffinity, name).Execute()
-	logger.V(2).Info("create volume", "name", name, "http", httpRes.Status)
+	logger := klog.FromContext(myclient.Ctx)
+
+	var response *client.VolumesObject
+	var httpRes *http.Response
+	var err error
+	if poolType == "Linear" {
```

Comment:
If you keep this, I would think you should have cases for an empty string (treat like "Linear" which should work on Virtual too?) and anything else (return an error).